video.cpp

```diff
@@ -3353,7 +3416,7 @@ void video_menu_bg(int n, int idle)
 			switch (n)
 			{
 			case 1:
-				if (!menubg) menubg = load_bg();
+				if (!menubg || cfg.game_wallpapers) menubg = load_bg();  // The OR statement allows reloading of the background image if the option is enabled.
```

Comment:
not sure what this does, but you don't want to reload the background if you didn't find a name for the entry if the old one is still displayed. I don't have a full grasp of the code and just from the diff i don't get it

Reply:
my question is, if i m displaying default.png and i have to load again default.png, what happens? is it smart enough to not doing anything?

Reply:
Ah... That is a good question and has value in reducing the operations if possible.  I'll look at that next to ensure that gets optimized.